controllers/console/plugin.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package console
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/node-healthcheck-operator/controllers/utils"
+	"github.com/openshift/api/console/v1alpha1"
+)
+
+const (
+	// PluginName is the name of the plugin and uses at several places
+	PluginName = "node-remediation-console-plugin"
+	// ServiceName is the name of the console plugin Service and must match the name of the Service in /bundle/manifests!
+	ServiceName = "node-healthcheck-operator-node-remediation-console-plugin"
+	// ServicePort is the port of the console plugin Service and must match the port of the Service in /bundle/manifests!
+	ServicePort = 9443
+)
+
+// +kubebuilder:rbac:groups=console.openshift.io,resources=consoleplugins,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch
+
+// CreateOrUpdateConsolePlugin creates or updates the resources needed for the remediation console plugin.
+// HEADS UP: consider cleanup of old resources in case of name changes or removals!
+//
+// TODO image reference to console plugin in CSV?
+func CreateOrUpdateConsolePlugin(ctx context.Context, mgr ctrl.Manager, namespace string, log logr.Logger) error {
+
+	// check if we are on OCP
+	if isOpenshift, err := utils.IsOnOpenshift(mgr.GetConfig()); err != nil {
+		errors.Wrap(err, "failed to check if we are on Openshift")
+	} else if !isOpenshift {
+		log.Info("we are not on Openshift, skipping console plugin activation")
+		return nil
+	}
+
+	// reader is needed because mgr isn't started yet, so caches are not filled yet and client.get fails
+	reader := mgr.GetAPIReader()
+	cl := mgr.GetClient()
+
+	// Create ConsolePlugin resource
+	// Deployment and Service are deployed by OLM
+	if err := createOrUpdateConsolePlugin(ctx, namespace, reader, cl); err != nil {
+		return err
+	}
+
+	log.Info("successfully created / updated console plugin resources")
+	return nil
+}
+
+func createOrUpdateConsolePlugin(ctx context.Context, namespace string, reader client.Reader, cl client.Client) error {
```

Comment:
Wouldn't it be simpler just to delete the old one and create the new one ?

Reply:
Mh, when we always delete and create, why does update exist at all? 😉 
I think the idea is that delete + create might cause much more overhead on the controllers which watch this resource, compared to an update.